src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1644 @@
+  expandCollapseCueSensitivity: 1 // sensitivity of expand-collapse cues
+}
+
+const expandCollapseOptionsCola = {
+  layoutBy: {
+    name: 'cose-bilkent',
+    animate: 'end',
+    randomize: false,
+    fit: false
+  },
+  // recommended usage: use cose-bilkent layout with randomize: false to preserve mental map upon expand/collapse
+  fisheye: false, // whether to perform fisheye view after expand/collapse you can specify a function too
+  animate: false, // whether to animate on drawing changes you can specify a function too
+  ready: function () {}, // callback when expand/collapse initialized
+  stop: function () {
+    this.layoutStopped = true
+    this.layoutReady = false
+    this.loading = false
+    this.loading = false
+    this.setHtmlLabel(this.cy)
+    if (tippy) {
+      tippy.hide()
+    }
+  },
+  undoable: true, // and if undoRedoExtension exists,
+  cueEnabled: true, // Whether cues are enabled
+  // expandCollapseCuePosition: 'top-left', // default cue position is top left you can specify a function per node too
+  expandCollapseCuePosition: function (ele) {
+    return ele.position()
+  },
+  expandCollapseCueSize: 20, // size of expand-collapse cue
+  expandCollapseCueLineSize: 16, // size of lines used for drawing plus-minus icons
+  expandCueImage: undefined, // image of expand icon if undefined draw regular expand cue
+  collapseCueImage: undefined, // image of collapse icon if undefined draw regular collapse cue
+  expandCollapseCueSensitivity: 1 // sensitivity of expand-collapse cues
+}
+
+export default {
+  name: 'Graph',
+  props: ['workflowid'],
+  data: function () {
+    return {
+      isConnected: false,
+      config,
+      elements,
+      graphData: {
+        nodes: [],
+        edges: []
+      },
+      i: 1,
+      // vue-spinner
+      color: '#5e9aff',
+      height: '35px',
+      width: '4px',
+      margin: '2px',
+      radius: '2px',
+      size: '.2em',
+      loading: true,
+      // layout variables
+      status: 'pending',
+      layoutName: 'dagre',
+      layoutStopped: true,
+      layoutReady: false,
+      freeze: false,
+      // elements
+      nodesEdges: [],
+      workflows: [],
+      edges: [],
+      subscriptions: {},
+      workflowId: '',
+      workflowOwner: '',
+      workflowName: '',
+      suiteName: ''
+    }
+  },
+  watch: {
+    graphData: debounce(function (newVal) {
+      this.debouncer.call()
+    }, 100),
+
+    layoutName: {
+      handler (value) {
+        this.updateLayout(value)
+      }
+    },
+
+    workflows: {
+      handler: function (newval, oldval) {
+        initialised ? this.workflowUpdated(newval) : console.debug('initialising')
+        initialised = true
+      },
+      deep: true
+    }
+  },
+
+  mixins: [mixin],
+
+  metaInfo () {
+    return {
+      title: this.getPageTitle('App.graph', { name: this.workflowId })
+    }
+  },
+
+  beforeRouteLeave (to, from, next) {
+    if (tippy) {
+      tippy.hide()
```

Comment:
This works now, but this is linking the component with the current route. Instead, I think we will have issues with this tooltip later.

If you look at the design sketch for the tabbed window:

![image](https://user-images.githubusercontent.com/304786/67731511-57981d00-fa5d-11e9-8c48-eebd63ff7e59.png)

It is possible to see that we may have multiple Graph components being used. Not all visible at the same time, though all existing in the same route/view. So the `beforeRouteLeave` would never be called.

Switching between Graphs (supposing we have 2 or more graphs open in the same panel), it could happen that a tippy tooltip would stay visible even after showing other graph (though it would depend on the z-index, and whether the issue happens when you have multiple graph components open).

That's why I had left a comment in a previous PR for the graph that the component should be completely separated from the route. Receiving everything it needed via parameters, and having methods that could be called when created/destroyed, or displayed/hidden, etc.

Probably a problem for later, but I think it would be a good idea to at least leave this comment so other reviewers will be aware of this (we had a bug fixed by Martin where the tippy tooltip would still be visible after you switched routes, from what I recall, that's why we have this `.hide()` call here).